sympy/logic/boolalg.py

```diff
@@ -2769,20 +2780,29 @@ def simplify_logic(expr, form=None, deep=True, force=False):
         made. By setting ``force`` to ``True``, this limit is removed. Be
         aware that this can lead to very long simplification times.
 
+    dontcare : Boolean expression
+        Optimize expression under the assumption that inputs where this
+        expression is true are don't care. This is useful in e.g. Piecewise
+        conditions, where later conditions do not need to consider inputs that
+        are convered by previous conditions. For example, if a previous
+        condition is ``And(A, B)``, the simplification of expr can be made
+        with don't cares for ``And(A, B)``.
+
     Examples
     ========
 
     >>> from sympy.logic import simplify_logic
     >>> from sympy.abc import x, y, z
-    >>> from sympy import S
     >>> b = (~x & ~y & ~z) | ( ~x & ~y & z)
     >>> simplify_logic(b)
     ~x & ~y
+    >>> simplify_logic(x | y, dontcare=y)
+    x
```

Comment:
I think a less trivial example would be better. Perhaps this:
```
In [6]: simplify_logic(x ^ y, dontcare=x & y)
Out[6]: x ∨ y
```
Here it is not as simple as just supposing that the dontcare is False.

Reply:
I can modify it as part of next PR. Agreed that it is probably better.